### DISCUSSION TITLE
Failure on build, resumable after redeploying. (managed by Lerna monorepo)

### DISCUSSION THREAD

**Discussion by happyeric77** [vercel/next.js#68095] [Help] [answered]
Hey team, 
I am coming across a weird state of deploying the project managed by Lerna.
Almost every time the update is pushed to `main` branch, the auto-deploy fails at build phase. But it will be resumed w/ no issue when I `redeploy` using Vercel dash board UI. like below:
![Screenshot 2024-07-24 at 16 57 56](https://github.com/user-attachments/assets/12c590ca-d90d-4aa1-a839-1231f2bb2565)

As mentioned, it always fails at build phase with the following error:
```bash
lib/context/NotifiTargetContext.tsx(469,10): error TS7006: Parameter 'tg' implicitly has an 'any' type.
lib/context/NotifiTargetContext.tsx(491,12): error TS7006: Parameter 'it' implicitly has an 'any' type.
lib/context/NotifiTargetContext.tsx(494,12): error TS7006: Parameter 'it' implicitly has an 'any' type.
lib/context/NotifiTargetContext.tsx(497,12): error TS7006: Parameter 'it' implicitly has an 'any' type.
lib/context/NotifiTargetContext.tsx(512,10): error TS7006: Parameter 'it' implicitly has an 'any' type.
lib/context/NotifiTargetContext.tsx(517,10): error TS7006: Parameter 'it' implicitly has an 'any' type.
lib/context/NotifiTargetContext.tsx(522,10): error TS7006: Parameter 'it' implicitly has an 'any' type.
Error: error occured in dts build
    at Worker.<anonymous> (/vercel/path0/node_modules/tsup/dist/index.js:2281:26)
    at Worker.emit (node:events:517:28)
    at MessagePort.<anonymous> (node:internal/worker:256:53)
    at [nodejs.internal.kHybridDispatch] (node:internal/event_target:786:20)
    at exports.emitMessage (node:internal/per_context/messageport:23:28)
 DTS Build error
```
The detail screenshot FYI herewith: 
![Screenshot 2024-07-24 at 17 02 35](https://github.com/user-attachments/assets/062f256d-3391-4f78-a473-f1f9d63db9e3)

Would like to ask a hint if the state above looks familiar to you.

Let me also share my project setting:

![Screenshot 2024-07-24 at 17 19 06](https://github.com/user-attachments/assets/d67cf44f-2695-4e4e-8f1f-e46404ae3d47)

**Comment by icyJoseph** [comment]
Oh wow, I am gonna totally guess, but I think the build pipeline has a cache too? Maybe, there are some types or other type of file declarations missing from your project, that get generated during the first attempt at build, but without them you can't build, so it fails. On the next run, they are available, and then the build passes.

Totally low effort guess... but maybe, clone your repository into another directory, and try to build it... does it also fail the first time?

**Reply by happyeric77** [reply]
Thanks for your feedback @icyJoseph.
If you are saying cloning the `lerna monorepo` and creating a new vercel project, yeah, I did. It behaves the same. Alwasy succeeded at the first build, but fails with those build coming afterward.

It makes sense about the cache. But I think the vercel should "rebuild" all dependencies repo as I config it like this.
![Screenshot 2024-07-24 at 17 35 24](https://github.com/user-attachments/assets/e7f55f7a-50e1-4338-b6dd-ca881416c514)
And looks like it does actually build those dependeny Repos successfully according to the log:

![Screenshot 2024-07-24 at 17 38 13](https://github.com/user-attachments/assets/0cdd1e03-4ec5-4e3a-96e6-d94d9dd5d2ca)

If that is really a cache issue, I am just wondering if we can config it somewhere to let us at least get this error around.

**Reply by icyJoseph** [reply]
I meant, locally, on your machine.

**Comment by happyeric77** [comment]
Hmmm .... thanks for giving the hint @icyJoseph.
looks like we have a way to bypass the build cache https://vercel.com/docs/deployments/troubleshoot-a-build#managing-build-cache
I will give it a try and feedback if it will still occur or not.

**Reply by happyeric77** [reply, marked as answer]
Hey, Just a quick update, I think adding the following env variable `VERCEL_FORCE_NO_BUILD_CACHE=1` fixes the issue.
